File: src/services/customerService.js
import { api } from "./api";

export async function listCustomers(params = {}) {
  return api.get("/customers", { params }).then((r) => r.data);
}

export async function getCustomerById(id) {
  return api.get(`/customers/${id}`).then((r) => r.data);
}

export async function createCustomer(payload) {
  return api.post("/customers", payload).then((r) => r.data);
}

export async function updateCustomer(id, payload) {
  return api.put(`/customers/${id}`, payload).then((r) => r.data);
}

export async function deleteCustomer(id) {
  return api.delete(`/customers/${id}`).then((r) => r.data);
}

File: backend/src/main/java/com/falconenergy/service/CustomerService.java
package com.falconenergy.service;

import com.falconenergy.dto.CustomerRequest;
import com.falconenergy.dto.CustomerResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface CustomerService {
    CustomerResponse createCustomer(CustomerRequest request);
    CustomerResponse getCustomerById(Long id);
    CustomerResponse updateCustomer(Long id, CustomerRequest request);
    void deleteCustomer(Long id);
    Page<CustomerResponse> getAllCustomers(String search, String status, Pageable pageable);
}
